app/src/cose.h

```diff
@@ -691,7 +498,7 @@ namespace scitt::cose
     // auxiliary buffer size.
     t_cose_parameters params;
     t_cose_sign1_verify_init(
-      &verify_ctx, T_COSE_OPT_TAG_REQUIRED | T_COSE_OPT_DECODE_ONLY);
+      &verify_ctx, T_COSE_OPT_TAG_REQUIRED | T_COSE_OPT_DECODE_ONLY | T_COSE_OPT_UNKNOWN_CRIT_ALLOWED);
```

Comment:
Given that this also affects non-Notary verification I think the `process_*_profile` functions in `verifier.h` should be extended to check that the crit field is empty for the other profiles. And there should be a comment here saying why using this flag is ok.

Reply:
Alternatively, add a flags argument to `cose::verify` that only the notary code path needs to set to `T_COSE_OPT_UNKNOWN_CRIT_ALLOWED`.